web/ui/src/common/resourcesFactory.js

```diff
@@ -269,16 +287,56 @@
 
             return function(/* args */){
                 var url = config.url.apply(null, arguments),
-                    payload;
+                    resourceName = url,
+                    payload,
+                    // deferred that will be returned to the user
+                    deferred = $q.defer();
+
+                // if resourceName has query params, strip em off
+                if(resourceName.indexOf("?")){
+                    resourceName = resourceName.split("?")[0];
+                }
+
+                // NOTE: all of our code expects angular's wrapped
+                // promises which provide a success and error method
+                // TODO - remove the need for this
+                httpify(deferred);
+
+                if(pendingGETRequests[resourceName]){
```

Comment:
Seems like this "if" statement should include a `&& config.method === "get"` so that if this callback is invoked for other types of methods, we don't reject them just because a GET is a little slow.